test/unit/lib/OneDev.test.js

```diff
@@ -0,0 +1,225 @@
+/* eslint-disable max-classes-per-file, no-shadow */
+
+import OneDev from '../../../source/lib/OneDev';
+import * as mockHelpers from '../../helpers';
+import User from '../../../source/lib/models/User';
+
+let req;
+
+const btoa = (contents) => Buffer.from(contents).toString('base64');
+
+beforeEach(() => {
+  jest.resetModules();
+  jest.restoreAllMocks();
+
+  req = mockHelpers.getMockRequest();
+});
+
+describe('OneDev interface', () => {
+  test('initialises the OneDev API wrapper', () => {
+    const onedev = new OneDev(req.params);
+
+    expect(onedev.api).toBeDefined();
+  });
+
+  describe('readFile', () => {
+    test('reads a file and returns its contents', () => {
+      const fileContents = 'This is a text file!';
+      const filePath = 'path/to/file.txt';
+      const mockGet = jest.fn(() =>
+        Promise.resolve({
+          body: {
+            base64Content: btoa(fileContents),
+          },
+        })
+      );
+
+      jest.mock('got', () => {
+        return {
+          extend: function mockGot() {
+            return {
+              get: mockGet,
+            };
+          },
+        };
+      });
+
+      const OneDev = require('../../../source/lib/OneDev').default;
+      const sut = new OneDev(req.params);
+
+      return sut.readFile(filePath).then((contents) => {
```

Comment:
Can use async/await throughout this file. Also, in any tests which make async calls we should add `expect.assertions(NUM_ASSERTIONS_IN_TEST)` prior to the async call so that we ensure any errors are caught.

https://stackoverflow.com/questions/50816254/necessary-to-use-expect-assertions-if-youre-awaiting-any-async-function-calls